trinity/tools/utils/async_generator_utils.py

```diff
@@ -0,0 +1,21 @@
+from typing import (
+    AsyncIterable,
+    Set,
+)
+
+
+async def contains_all(async_gen: AsyncIterable[str], keywords: Set[str]) -> bool:
```

Comment:
Not sure if this file should rather live in `trinity/utils` as it really isn't directly coupled to `trinity/tools` in any way other then `trinity/tools` using it. That said, not sure how often I caught myself thinking "Damn, all I wish for was a function that takes an `AsyncIterable[str]` and returns a boolean if it contains all of some given keywords"....probably never :thinking:

Reply:
I don't think `trinity.tools` needs its own `utils` module.  I'm good with this living under `trinity.utils.async` or something similar.